File: LeetCode/3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        nums_set = set(nums)
        dummy = tail = ListNode()
        current_node = head
        while current_node is not None:
            if current_node.val not in nums_set:
                tail.next = current_node
                tail = tail.next
            current_node = current_node.next

        tail.next = None
        return dummy.next
        

File: LeetCode/3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> nums_set(nums.begin(), nums.end());
        ListNode *dummy = NULL, *prev = NULL;
        dummy = prev = new ListNode();
        dummy->next = NULL;

        while(head != NULL){
            if(nums_set.find((head->val)) == nums_set.end()){
                prev->next = head;
                prev = head;
            }
            head = head->next;
        }

        prev->next = NULL;
        return dummy->next;
    }
};